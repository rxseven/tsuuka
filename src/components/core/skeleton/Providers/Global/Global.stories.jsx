import React from 'react';
import { storiesOf } from '@storybook/react';

import mock from 'stories/mock';
import notes from 'stories/shared/notes';
import Global from './index';

const { content } = mock.elements;
const title = 'Components/Core/Skeleton/Providers/Global';
const component = <Global>{content}</Global>;

storiesOf(title, module).add('default', () => component, {
  notes: notes.provider
});
