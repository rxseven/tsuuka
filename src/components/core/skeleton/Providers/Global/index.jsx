/* eslint-disable react/no-unused-state */
import PropTypes from 'prop-types';
import exact from 'prop-types-exact';
import React from 'react';

const initialState = { isMenu: false };
const Context = React.createContext({});

const propTypes = exact({
  children: PropTypes.element.isRequired
});

class Global extends React.Component {
  state = { ...initialState };

  handleMenuChange = state => {
    this.setState({ isMenu: state.isOpen });
  };

  handleMenuClose = () => {
    this.setState({ isMenu: false });
  };

  handleMenuToggle = () => {
    this.setState(state => ({ isMenu: !state.isMenu }));
  };

  render() {
    const { children } = this.props;
    const actions = {
      menu: {
        onChange: this.handleMenuChange,
        onClose: this.handleMenuClose,
        onToggle: this.handleMenuToggle
      }
    };
    const { state } = this;

    return (
      <Context.Provider value={{ actions, state }}>{children}</Context.Provider>
    );
  }
}

Global.propTypes = propTypes;

export { Context };
export default Global;
