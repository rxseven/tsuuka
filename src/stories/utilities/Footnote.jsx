import React from 'react';
import styled from 'styled-components';

const Frame = styled.div`
  bottom: 2rem;
  font-size: 14px;
  position: fixed;
`;

function Footnote({ children }) {
  return <Frame>{children}</Frame>;
}

export default Footnote;
