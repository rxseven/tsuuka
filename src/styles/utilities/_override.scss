// Reset button's default styles
button:focus {
  box-shadow: none !important;
  outline: 0 !important;
}
