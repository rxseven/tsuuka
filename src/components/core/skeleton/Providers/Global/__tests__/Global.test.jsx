import React from 'react';

import { factory, toMarkup } from 'tests/utilities';
import Global from '../index';

// Arrange
const seed = { content: 'content' };
const source = { children: <span>{seed.content}</span> };
const input = { ...seed, ...source };

// Setup
function setup(props) {
  return factory(Global, source, props);
}

// Test suites
describe('<Providers.Global />', () => {
  it('should render without crashing', () => {
    setup();
  });

  it('should render passed children correctly', () => {
    const expected = {
      content: input.content,
      html: input.children
    };
    const { component } = setup();

    expect(component).toHaveTextContent(expected.content);
    expect(component).toContainHTML(toMarkup(expected.html));
  });
});
