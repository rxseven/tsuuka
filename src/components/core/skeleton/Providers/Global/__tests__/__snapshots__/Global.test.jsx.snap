// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Global /> Snapshot tests should render correctly 1`] = `
<div>
  Content
</div>
`;
